recipes/pydantic-core/0001-build-Use-PYTHON-to-determine-the-correct-python-exe.patch

```diff
@@ -0,0 +1,31 @@
+From fc6e90e872badd8721ea84e40da1e8561dc52674 Mon Sep 17 00:00:00 2001
+From: "Uwe L. Korn" <uwe.korn@quantco.com>
+Date: Wed, 22 Mar 2023 16:46:25 +0100
+Subject: [PATCH] build: Use `PYTHON` to determine the correct python
+ executable
+
+---
+ build.rs | 3 ++-
+ 1 file changed, 2 insertions(+), 1 deletion(-)
+
+diff --git a/build.rs b/build.rs
+index 4518799..a2679f4 100644
+--- a/build.rs
++++ b/build.rs
+@@ -1,3 +1,4 @@
++use std::env;
+ use std::path::Path;
+ use std::process::Command;
+ use std::str::from_utf8;
+@@ -10,7 +11,7 @@ fn generate_self_schema() {
+         // skip running generate_self_schema.py
+         return;
+     }
+-    let output = Command::new("python3")
++    let output = Command::new(env::var("PYTHON").unwrap_or_else(|_| "python3".to_owned()))
```

Comment:
is there a reason why conda requires using `PYTHON` instead of adding the correct `python3` to `PATH`?

Reply:
There is no `python3` on Windows. Seems like also other people have tripped over that: https://github.com/pydantic/pydantic-core/issues/355

Reply:
This will also be relevant for cross-compiling to ensure the correct python executable is picked up. Cross-compiling is not yet tested on `staged-recipes`, though.